ftd/t/js/80-or-type-constant.ftd

```diff
@@ -11,6 +11,15 @@
 
 -- button-type b: small
 
+;; A variable with the same name as a variant of a button type
+;; if we comment this out, the value of "small" will be inferred from the right-hand side or-type
+-- integer small: 2
+
+;; Since the specificity of a variable is higher, the value of the variable will be used
+;; and !smol will be printed
+-- ftd.text: !Smol
+text if { b == small }: Smol
```

Comment:
@harshdoesdev This must fail, the type of `b` is `button-type` and `small` is `integer`. You can not compare different types. Make sure this fails, and then write another test case where `small` is of `button-type` type, and then do the equality check.

Reply:
@amitu this issue has been resolved and I have tested it locally, now it will throw an error if an integer/any-other-value is provided as the value on either side of equality check.

Reply:
Sorry, one test is still failing